src/Teams/Controllers/ProgramTestController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Teams.Data;
+using Teams.Data.ProgramTestRepos;
+using Teams.Data.Repositories;
+using Teams.Domain;
+using Teams.Models;
+
+namespace Teams.Controllers
+{
+    public class ProgramTestController : Controller
+    {
+        private  IProgramCodeQuestionRepository _questionRepository;
+        private IProgramTestRepository _testRepository;
+        private ApplicationDbContext _context;
+
+        public ProgramTestController(IProgramCodeQuestionRepository questionRepository, IProgramTestRepository testRepository, ApplicationDbContext context)
+        {
+            _context = context;
```

Comment:
Please, avoid using context directly. Move all necessary methods of context to the repository.